twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -291,6 +301,23 @@ public void run() {
     // Since all the runnables are now stopped, it is okay to stop the poller.
     stopPoller.shutdownNow();
     cleanupDir();
+    if (eventHandler != null) {
+      if (finalStatus == null) {
+        // if finalStatus is not set, the application must be stopped by a SystemMessages#STOP_COMMAND
+        eventHandler.killed(twillRuntimeSpec.getTwillAppName(), twillRuntimeSpec.getTwillAppRunId());
+      }
+      switch (finalStatus) {
+        case COMPLETED:
+          eventHandler.completed(twillRuntimeSpec.getTwillAppName(), twillRuntimeSpec.getTwillAppRunId());
+          break;
+        case ABORTED:
+        eventHandler.aborted(twillRuntimeSpec.getTwillAppName(), twillRuntimeSpec.getTwillAppRunId());
+          break;
+        default:
+          // should never reach here
+          LOG.error("Unsupported FinalStatus '%s'", finalStatus.name());
```

Comment:
That's not the right syntax. slf4j logger uses `{}`. Should be `LOG.error("Unsupported status {}", finalStatus)`.